frontend/src/cljs/imas_seamap/map/views.cljs

```diff
@@ -202,7 +202,13 @@
                                   {:title "Server Error"
                                    :icon  "error"}]
     ;; Default; we have actual content:
-    [:div {:dangerouslySetInnerHTML {:__html info-body}}]))
+    [:div
+     {:ref
+      (fn [element]
+        (when element
+          (set! (.-innerHTML element) nil)
+          (doseq [{:keys [_body _style] :as response} responses]
+            (.appendChild element (create-shadow-dom-element response)))))}]))
```

Comment:
What's the deal with refs, out of curiosity -- the only thing I'd want to ensure here is that the function isn't going to be called repeatedly (which I assume it doesn't, I've just never looked into it)

Reply:
Once per state change, at least that's what my testing shows.